modules/casper.js

```diff
@@ -2513,6 +2513,7 @@ Casper.prototype.newPage = function newPage() {
     if (utils.isObject(this.options.viewportSize)) {
         this.page.viewportSize = this.options.viewportSize;
     }
+    this.page.clearMemoryCache()
```

Comment:
It worked for me, running in Linux/Ubuntu/Xenial, with phantomjs 1.98, 2.1.1 and 2.5:

henrique@usb-azul:~/test_casper$ lsb_release -a ; PATH=$PATH:./phantomjs/ ; echo "casper.js with this.page.clearMemoryCache():" ; grep -B16 -A3 MemoryCache ./casperjs/modules/casper.js ; echo "My test script" ; cat ./test.js ; cp phantomjs/phantomjs-250 phantomjs/phantomjs ; phantomjs --version ; ./casperjs/bin/casperjs ./test.js ; cp phantomjs/phantomjs-211 phantomjs/phantomjs ; phantomjs --version ; ./casperjs/bin/casperjs ./test.js ; cp phantomjs/phantomjs-198 phantomjs/phantomjs ; phantomjs --version ; ./casperjs/bin/casperjs ./test.js 
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
casper.js with this.page.clearMemoryCache():
* Allow user to create a new page object after calling a casper.page.close()
* @return WebPage
*/

Casper.prototype.newPage = function newPage() {
    "use strict";
    this.checkStarted();
    this.page.close();
    this.page = this.mainPage = createPage(this);
    this.page.settings = utils.mergeObjects(this.page.settings, this.options.pageSettings);
    if (utils.isClipRect(this.options.clipRect)) {
        this.page.clipRect = this.options.clipRect;
    }
    if (utils.isObject(this.options.viewportSize)) {
        this.page.viewportSize = this.options.viewportSize;
    }
    this.page.clearMemoryCache()
    return this.page;
};

My test script

var casper = require('casper').create({
	verbose: true,
	logLevel: "debug",
});

casper.start("");
casper.thenOpen("http://www.google.com/", function () { this.capture("test1.png"); });
casper.then( function () { this.page = this.newPage(); });
casper.thenOpen("http://www.yahoo.com", function () { this.capture("test2.png"); });
casper.run();
2.5.0-development
[info] [phantom] Starting...
[info] [phantom] Running suite: 5 steps
[debug] [phantom] opening url: http://www.google.com/, HTTP GET
[debug] [phantom] Navigation requested: url=http://www.google.com/, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "http://www.google.com/"
[debug] [phantom] Successfully injected Casper client-side utilities
[info] [phantom] Step anonymous 2/5 http://www.google.com/ (HTTP 0)
[debug] [phantom] Capturing page to /home/henrique/test_casper/test1.png
[info] [phantom] Capture saved to /home/henrique/test_casper/test1.png
[info] [phantom] Step anonymous 2/5: done in 3403ms.
[info] [phantom] Step anonymous 3/5 http://www.google.com/ (HTTP 0)
[info] [phantom] Step anonymous 3/5: done in 3421ms.
[debug] [phantom] opening url: http://www.yahoo.com/, HTTP GET
[debug] [phantom] Navigation requested: url=http://www.yahoo.com/, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "http://www.yahoo.com/"
[debug] [phantom] Navigation requested: url=https://s.yimg.com/os/mit/media/m/base/images/transparent-95031.png, type=Other, willNavigate=true, isMainFrame=false
[debug] [phantom] Successfully injected Casper client-side utilities
[info] [phantom] Step anonymous 5/5 http://www.yahoo.com/ (HTTP 0)
[debug] [phantom] Capturing page to /home/henrique/test_casper/test2.png
[info] [phantom] Capture saved to /home/henrique/test_casper/test2.png
[info] [phantom] Step anonymous 5/5: done in 7298ms.
[info] [phantom] Done 5 steps in 7366ms
[debug] [phantom] Navigation requested: url=about:blank, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "about:blank"
2.1.1
[info] [phantom] Starting...
[info] [phantom] Running suite: 5 steps
[debug] [phantom] opening url: http://www.google.com/, HTTP GET
[debug] [phantom] Navigation requested: url=http://www.google.com/, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] Navigation requested: url=http://www.google.com.br/?gws_rd=cr&ei=veHHWO3fEMa-wATeyLTgAw, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "http://www.google.com.br/?gws_rd=cr&ei=veHHWO3fEMa-wATeyLTgAw"
[debug] [phantom] Successfully injected Casper client-side utilities
[info] [phantom] Step anonymous 2/5 http://www.google.com.br/?gws_rd=cr&ei=veHHWO3fEMa-wATeyLTgAw (HTTP 200)
[debug] [phantom] Capturing page to /home/henrique/test_casper/test1.png
[info] [phantom] Capture saved to /home/henrique/test_casper/test1.png
[info] [phantom] Step anonymous 2/5: done in 1622ms.
[info] [phantom] Step anonymous 3/5 http://www.google.com.br/?gws_rd=cr&ei=veHHWO3fEMa-wATeyLTgAw (HTTP 200)
[info] [phantom] Step anonymous 3/5: done in 1628ms.
[debug] [phantom] opening url: http://www.yahoo.com/, HTTP GET
[debug] [phantom] Navigation requested: url=http://www.yahoo.com/, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] Navigation requested: url=https://www.yahoo.com/, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] Navigation requested: url=https://br.yahoo.com/?p=us, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "https://br.yahoo.com/?p=us"
[debug] [phantom] Navigation requested: url=https://s.yimg.com/os/mit/media/m/base/images/transparent-95031.png, type=Other, willNavigate=true, isMainFrame=false
[debug] [phantom] Successfully injected Casper client-side utilities
[info] [phantom] Step anonymous 5/5 https://br.yahoo.com/?p=us (HTTP 200)
[debug] [phantom] Capturing page to /home/henrique/test_casper/test2.png
[info] [phantom] Capture saved to /home/henrique/test_casper/test2.png
[info] [phantom] Step anonymous 5/5: done in 4424ms.
[info] [phantom] Done 5 steps in 4485ms
[debug] [phantom] Navigation requested: url=about:blank, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "about:blank"
1.9.8
[info] [phantom] Starting...
[info] [phantom] Running suite: 5 steps
[debug] [phantom] opening url: http://www.google.com/, HTTP GET
[debug] [phantom] Navigation requested: url=http://www.google.com/, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] Navigation requested: url=http://www.google.com.br/?gws_rd=cr&ei=wuHHWJDIAoajwAT-toTYBg, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "http://www.google.com.br/?gws_rd=cr&ei=wuHHWJDIAoajwAT-toTYBg"
[debug] [phantom] Successfully injected Casper client-side utilities
[info] [phantom] Step anonymous 2/5 http://www.google.com.br/?gws_rd=cr&ei=wuHHWJDIAoajwAT-toTYBg (HTTP 200)
[debug] [phantom] Capturing page to /home/henrique/test_casper/test1.png
[info] [phantom] Capture saved to /home/henrique/test_casper/test1.png
[info] [phantom] Step anonymous 2/5: done in 1519ms.
[info] [phantom] Step anonymous 3/5 http://www.google.com.br/?gws_rd=cr&ei=wuHHWJDIAoajwAT-toTYBg (HTTP 200)
Unsafe JavaScript attempt to access frame with URL about:blank from frame with URL file:///home/henrique/test_casper/casperjs/bin/bootstrap.js. Domains, protocols and ports must match.